collections.mdx

```diff
@@ -132,29 +195,192 @@ The following types are supported:
+}
+```
 
 <Info>Additional types such as dates will be added soon.</Info>
 
-#### Optional Fields
+#### Primitives
 
-All fields are required by default, if you want to make a field optional simply
-append `?` after the field name and before the `:`. For example:
+The following primitive types are supported:
+
+ - `string`
+ - `number`
+ - `boolean`
+
+#### Bytes
+
+The bytes type allows you to store arbitary bytes in Polybase, without having to encode the data as a string. 
+
+If you use our client library [`@polybase/client`](https://www.npmjs.com/package/@polybase/client), the bytes type will be returned as a `Uint8Array`. If this field is returned
+from the [Gateway API](/api-reference/overview), then it will be a base64 encoded string.
+
+<Note>
+`bytes` data is currently not indexed
+</Note>
+
+
+#### Collection
+
+`Collection` fields reference another collection in the same namespace. This allows you to create relationships between records in the same or different collections. 
+Similar to how you have relationships in SQL.
+
+You can define the field on the collection, and then also pass a parameter as a `Collection` type.
 
 ```js
-@public
-collection CollectionName {
+@public Pet {
   id: string;
-  required: number;
-  optional?: number;
 
+  // This field is referencing the User collection below
+  owner: User;
+
+  constructor (id: string, owner: User) {
+    this.id = id;
+    this.owner = owner;
+  }
+}
+
+@public
+collection User {
+  id: string;
   ...
 }
 ```
 
+You can then call the constructor using `create()` passing in the reference to a specific record of that collection:
+
+<CodeGroup>
+```js js
+import { Polybase } from "@polybase/client";
+
+const db = new Polybase({
+  defaultNamespace: "your-namespace",
+});
+
+db.collection('Pet').create(
+  [
+    "pet1", 
+    // This is referencing the User collection
+    db.collection('User').record("user1")
+  ]
+);
+```
+
+```bash curl
+curl -X POST \
+  https://testnet.polybase.xyz/v0/collections/your_namespace%2FPet/records \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "args": [
+      "pet1",
+      {
+        "collectionId": "your_namespace/Pet",
+        "id": "user1"
+      }
+    ]
+}'
+```
+</CodeGroup>
+
+
+#### Public Key
+
+The `PublicKey` type is a special type that is used to represent a public key. Currently, we only support Ethereum (`secp256k1`) public keys.
+
+**`.toHex()`**
+
+If you need a string hex representation of the public key, you can use `this.publicKey.toHex()`. This will return a string starting with `0x`, 
+and the hexidecimal representation of the 64 byte public key.
+
+<Note>
+  Public keys on Polybase use non-prefixed Public Keys, so they are 64 bytes long.
+</Note>
+
+<Info>We will be adding support for [Falcon](https://eprint.iacr.org/2022/1041.pdf), this is a zk optimized PublicKey</Info>
+
+#### Arrays
+
+You can create arrays of `string`, `number`, `boolean`, `PublicKey` and `Collection` type. For example:
```

Comment:
```suggestion
You can create arrays of `string`, `number`, `boolean`, `bytes` and `Collection` type. For example:
```

`PublicKey[]` is not supported right now. `bytes[]` is allowed.

Reply:
Ah okay, will create a ticket. Workaround right now would be to delegate to User[] then.